### PR TITLE
kapp-controller: v0.30.2

### PR DESCRIPTION
**Pull request by joe-kimmel-vmw**
## What this PR does / why we need it

Adds kapp-controller 0.30.2 LTS release

## Describe testing done for PR
visually checked output of ytt render step.

### CODE REVIEW

File: addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: Bundle
+metadata:
+  name: kapp-controller
+authors:
+- name: Daniel Guo
+  email: gdaniel@vmware.com
+- name: Lucheng Bao
+  email: luchengb@vmware.com
```
[the remark this answers is not in the document]
Comment (joe-kimmel-vmw) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: @seemiller  is there some evergreen value we could put in here, like `carvel@vmware.com` ?    What are the goals of this field -- are we imagining that someone is going to reach out via email?   Wouldn't we prefer them to come to our slack channel?

[the remark this answers is not in the document]
Comment (joe-kimmel-vmw) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: e.g. on[ kc 34.x line ](https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.34.0/bundle/.imgpkg/bundle.yml#L5)we also haven't been maintaining this metadata - who is the audience for this info (and is emailing any single team member an effective means of contacting us?)

[the remark this answers is not in the document]
Comment (seemiller) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: I think that the current direction is to omit the email. If you feel that listing the maintainers in the package is overkill in addition to the metadata.yaml, I would simply remove the list from the bundle.yml file.

[the remark this answers is not in the document]
Comment (joe-kimmel-vmw) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: I'm happy to remove emails, and even the entire 'authors' from the package.   But even the 'maintainers' list  seems not just redundant to what's provided via `git log` / github contributors UX,  but more likely to be out of date than checking the provided website (which has preferred contact methods such as slack channels, and a list of team member names that can be updated dynamically).  

 I'm trying to understand what the driver is for including this information anywhere in any form. Appreciating there must've been a discussion at some point and I obviously wasn't there, but who is the audience for this? What are the goals of including and maintaining a list of names?

[the remark this answers is not in the document]
Comment (joe-kimmel-vmw) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: Also - unclear what to push to for the modified metadata:
```
imgpkg: Error: Writing 'projects.registry.vmware.com/tce/kapp-controller:0.30.2':
  Writing image:
    PUT https://projects.registry.vmware.com/v2/tce/kapp-controller/manifests/0.30.2:
      PRECONDITION:
        Failed to process request due to 'kapp-controller:0.30.2' configured as immutable.
```

should I make a new kc-0.30.2.1 that only exists in TCE?

File: addons/packages/kapp-controller/0.30.2/package.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: kapp-controller.community.tanzu.vmware.com.0.30.1
+spec:
+  refName: kapp-controller.community.tanzu.vmware.com
+  version: 0.30.1
+  releaseNotes: "kapp-controller 0.30.1 https://github.com/vmware-tanzu/carvel-kapp-controller"
```
Comment (seemiller) [inline addons/packages/kapp-controller/0.30.2/package.yaml:8]:
```suggestion
  name: kapp-controller.community.tanzu.vmware.com.0.30.2
spec:
  refName: kapp-controller.community.tanzu.vmware.com
  version: 0.30.2
  releaseNotes: "kapp-controller 0.30.2 https://github.com/vmware-tanzu/carvel-kapp-controller"
```

**Review by seemiller — CHANGES_REQUESTED** [review]

File: addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joe-kimmel-vmw) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: just to include in this thread: https://github.com/vmware-tanzu/community-edition/pull/3890/files for longer-term evergreen-ness.   and added "3.) edit or remove the authors list, if present, from addons/packages/kapp-controller/<your new version>/bundle/.imgpkg/bundle.yml".  to our internal instructions for this operation.

[the remark this answers is not in the document]
Comment (seemiller) [reply addons/packages/kapp-controller/0.30.2/bundle/.imgpkg/bundle.yml:9]: re: maintainers - just wanting to keep the metadata current and accurate.

re: push/immutable tag - checked the package, the SHA is good!

**Review by seemiller — APPROVED** [review]

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Pull request merged**